src/main/java/uk/gov/hmcts/reform/roleassignment/controller/advice/RoleAssignmentControllerAdvice.java

```diff
@@ -60,6 +64,49 @@
         );
     }
 
+    @ExceptionHandler(HttpMessageNotReadableException.class)
+    public ResponseEntity<ErrorResponse> notReadableException(final HttpMessageNotReadableException e) {
+        return deserializeError(e, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(NullPointerException.class)
+    public ResponseEntity<ErrorResponse> nullException(final NullPointerException e) {
+        return nullError(HttpStatus.BAD_REQUEST);
```

Comment:
It looks weird as you are handling null pointer exception and returning Bad Request.

Reply:
I just thought because when we validate the inputs for null we return bad request as well.

public static void validateInputParams(String pattern, String inputString) {
        if (StringUtils.isEmpty(inputString)) {
            throw new BadRequestException("An input parameter is Null/Empty");
...
}